creator-node/src/services/stateMachineManager/stateReconciliation/SecondarySyncHealthTracker.ts

```diff
@@ -1,332 +1,284 @@
-): Promise<string[]> {
-  const stream = redisClient.scanStream({ match: pattern })
-
-  const keySet = new Set<string>()
-  return new Promise<string[]>((resolve, reject) => {
-    stream.on('data', async (keys = []) => {
-      keys.filter(extraFilter).forEach((key) => {
-        keySet.add(key)
+      // If under, record retry to 'true'. Else, record retry as 'false'.
+      this._updateWalletToSecondaryToShouldContinueAction({
+        shouldContinueAction,
+        wallet,
+        secondary
       })
-    })
-    stream.on('end', () => {
-      resolve(Array.from<string>(keySet).filter(Boolean))
-    })
-    stream.on('error', (e: any) => {
-      reject(e)
-    })
-  })
-}
+    }
+  }
 
-/**
- * Builds redis key pattern given params, using today as the default date
- * and wildcard matcher for every other default param.
- * Key pattern string can map to one or multiple keys.
- */
-function _getRedisKeyPattern({
-  secondary = '*',
-  wallet = '*',
-  syncType = '*',
-  outcome = '*',
-  date = null
-}: RedisFilters) {
-  // format: YYYY-MM-DD
-  date = date || new Date().toISOString().split('T')[0]
-
-  return `${RedisKeyPrefix}:::${secondary}:::${wallet}:::${syncType}:::${date}:::${outcome}`
-}
+  _updateWalletToSecondaryToShouldContinueAction({
+    shouldContinueAction,
+    wallet,
+    secondary
+  }: {
+    wallet: string
+    secondary: string
+    shouldContinueAction: boolean
+  }) {
+    if (!this.walletToSecondaryToShouldContinueAction[wallet]) {
+      this.walletToSecondaryToShouldContinueAction[wallet] = {}
+    }
 
-function _parseRedisKeyIntoComponents(key: string) {
-  const components = key.split(':::')
-  const [, secondary, wallet, syncType, date, outcome] = components
-  return { secondary, wallet, syncType, date, outcome }
-}
+    this.walletToSecondaryToShouldContinueAction[wallet][secondary] =
+      shouldContinueAction
+  }
 
-async function _recordSyncRequestOutcome(
-  secondary: string,
-  wallet: string,
-  syncType: string,
-  success = true
-) {
-  try {
-    const outcome = success ? Outcomes.SUCCESS : Outcomes.FAILURE
-    const redisKey = _getRedisKeyPattern({
-      secondary,
-      wallet,
-      syncType,
-      outcome
-    })
+  _shouldContinueAction(error: string, errorCount: number) {
+    const maxRetries = SYNC_ERRORS_TO_MAX_NUMBER_OF_RETRIES_MAP.get(error)
 
-    // incr() will create key with value 0 if non-existent
-    await redisClient.incr(redisKey)
+    if (maxRetries === undefined) {
+      return errorCount <= SYNC_ERRORS_TO_MAX_NUMBER_OF_RETRIES.default
+    }
 
-    // Set key expiration time (sec) in case it hasn't already been set (prob not most efficient)
-    await redisClient.expire(redisKey, DailyRedisKeyExpirationSec)
+    return errorCount <= maxRetries
+  }
 
-    logger.debug(
-      `SecondarySyncHealthTracker:_recordSyncRequestOutcome || Recorded ${redisKey}`
-    )
+  _getSyncFailureRedisKey({
+    secondary,
+    wallet,
+    syncType
```

Comment:
did you end up finding a need for syncType after our last chat? otherwise we should remove it

Reply:
nah we dont need it. can remove